### ISSUE TITLE
[Feature Request] Changes to default values

### ISSUE THREAD

**Issue by WesCook** [mythridium/combat-simulator#44]
Combat Sim is a fantastic utility, and the updates for Melvor 1.3 are very nice.  I really appreciate how quickly you were able to update it, even with all the changes to Modifiers.

I'd like to make a (hopefully) low-hanging fruit request to streamline usage even more.  Currently when launching the Combat Sim, I always do three things:

1. Go to the Equipment tab, import my current loadout.
2. Go to the Simulate tab, click "Select Active Target".
3. Change Plot Type to "Kills" or "Drop Rate".

Then I actually run the simulation, and start making tweaks as necessary.

While Combat Sim offers a lot of customization options, I think these changes would save more players time without being too confusing or "opinionated" of defaults.  I'll offer a justification for each change.

**Import current loadout**

While it isn't possible to infer the specific numbered equipment set, it does make sense to import the currently equipped loadout when opening the tool.  This is actually preferred, as you may have a slightly tweaked loadout if adding something like a Slayer item to make combat legal.  In those situations, it would require importing your closest loadout and then tweaking the build to match your player.

I expect that in most occasions, players would want to start with their currently-equipped loadout and test changes from there.

As a side note, having a `[Current]` button beside `[1][2][3]` when importing equipment sets might be useful as well.

**Select Active Target**

Essentially, if a player is already engaged in combat, they probably want to focus on that specific situation.  The target should be automatically selected and the appropriate combat category (Dungeons, Slayer, etc) toggled on if necessary.

**Plot Type**

Right now the default plot type is just the top entry: XP per [time].  I would argue this is not an especially helpful metric.

While I think "Drops" is very useful, that's not universally the case.  Particularly in situations where you're farming a rare drop, but the monster has many common drops as well.  So I'd suggest that "Kills" is the most universally helpful, and a good default.

It might be that plot type is intended for comparing values across a number of different targets, rather than testing different loadouts against one.  However I often still use it with even a single target.  It makes it easier to focus on a specific metric, without needing to find it in the sidebar each time.

Thank you for the mod, and for your consideration on these changes.  Please let me know if I can offer clarification on any of these points.

**Comment by mythridium** [comment]
Thank you so much for your feedback, it is very valuable.

**Import current loadout** - this would be fairly annoying for some people, myself included as I work on combat sim, it's nice to have a fresh slate on startup, particularly for bug finding, and could also be annoying if I have my skilling set on. However, I can see the benefit, so I've added a new setting which will automatically import your current equipment set when combat sim startups on game load. This lets you toggle it on/off as you want it.

**Select active target** - Personally I'm usually not in combat with something before I've simmed them, I would typically sim before I get into combat with something, so having it shift my target on open would be annoying. I haven't seen this one asked for before,  however I'm open to changes if I get more feedback on this. I'll keep an eye out if this comes up more, but for the moment I'll leave is, hope you understand.

**Plot Type** - I've made a change which should address this, the plot skill, type and time dropdowns are now persisted across all characters and devices. It will remember your last selection, so if you load up combat sim it will be the same as where you left off.

Thank you again for your feedback.

Fixed in v3.0.69 c945baa

**Comment by WesCook** [comment]
I appreciate the swift response and even an update.  You must feel the same way about leaving open Issues hanging around as I do.  ;)

I understand that my preferred defaults may vary from others, so it's no problem at all if you wish to leave some things the way they are.  If it's easier for development this way, or you think it's just better suited for the general public, that's completely fair.

> Import current loadout - "However, I can see the benefit, so I've added a new setting which will automatically import your current equipment set when combat sim startups on game load. This lets you toggle it on/off as you want it."

That's great, thank you.  I realize I wasn't actually sure about when the initial settings were loaded; if that was on game load or first open of the tool.  Hopefully the new setting wasn't too much trouble.  If you prefered the solution of clicking a button to import the current loadout without any "magic" behind the scenes, I think that'd also be fine.  Will try out the new setting now though.

> Select active target - "I haven't seen this one asked for before, however I'm open to changes if I get more feedback on this."

Sure thing.  I am a little surprised it hasn't come up before, but maybe more players use it outside of combat the way you do.

I completely understand the rationale about not wanting it to automatically change settings when opened.  If it were implemented that might require another setting, and at a certain point that makes the tool harder to use and develop.  It's tough to find the right balance in functionality and ease-of-use sometimes.  But it's also why I'm fine with developers saying "no" to new features.  Everything has a cost.

> Plot Type - "It will remember your last selection"

An even better solution.  Good call.

Thanks again for your consideration and the quick implementation.  Best of luck in the future.

**Comment by WesCook** [comment]
Is the new setting called "Import On Startup"?  If so, I'm afraid to say that it doesn't actually appear to work in its current form.  It was enabled by default, but when I load into the game my loadout is still empty.

If this is an unrelated setting or I've misunderstood, please let me know.  I can also file a separate bug report, or attach my save file if that would be helpful.  Thank you.

**Comment by mythridium** [comment]
@WesCook Thanks for giving it a go, yeah looks I messed up the default value when nothing has been persisted yet, it should be off by default.

I've released a fix for this in v3.0.70.

Also, after thinking about it more, I think it makes more sense if I shift it to import on first open, might be confusing for someone who starts playing and only opens combat sim an hour into playing with the setting on, the import would be an hour behind.

I've made this change and renamed the setting from 'Import on startup' to 'Import on first open'

Let me know if this doesn't resolve your issue.
